includes/Taxonomy/Taxonomy_Base.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Class Taxonomy_Base.
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Taxonomy;
+
+use Google\Web_Stories\Infrastructure\PluginDeactivationAware;
+use Google\Web_Stories\Infrastructure\SiteInitializationAware;
+use Google\Web_Stories\Service_Base;
+use WP_Site;
+
+/**
+ * Taxonomy_Base class
+ */
+abstract class Taxonomy_Base extends Service_Base implements PluginDeactivationAware, SiteInitializationAware {
```

Comment:
Should we migrate `Media_Source_Taxonomy` to extend this class? I see you did it in 26d486344c7d1cdec3f4e9375f2384577d1ec5c4, but somehow reverted it again. What's the reason?

Should it at least use `Stories_Terms_Controller`?

Reply:
I think this should be done in a follow up PR. There was more too it than I thought, so I had to revert it.